### PR TITLE
moves from exec to spawnsync and this removes the buffer outage probl…

### PR DESCRIPTION
**Pull request by nilaysaha**
Also added a bash shell that automates the various steps.

### CODE REVIEW

File: cardanoLeaderLogs.js [hunk 1]
```diff
@@ -24,7 +24,8 @@ if(process.argv.length >= 6 && !isNaN(parseFloat(process.argv[5]))) {
   overwriteDFactor = parseFloat(process.argv[5])
 }
 
-const params                  = JSON.parse(fs.readFileSync(process.argv[2]))
+console.log("trying to read file"+process.argv[3])
+const params = JSON.parse(fs.readFileSync(process.argv[3]))
```
Comment (dostrelith678) [inline cardanoLeaderLogs.js:28]: why this change? The usage is:

`node cardanoLeaderLogs.js config.json epochNonce`

so config is argv[2]

File: myleaderlog.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+echo "Please note that this relies on having a cloned version of:https://github.com/DamjanOstrelic/cardano-leader-logs."
+echo "Then change the directory where this cloning is done into CARDANO_LLOG_EXEC parameter"
+echo "Also we need to have vrf.skey into the VRF_KEY_FILE parameter"
+
+
+SLOTLEADER_CONFIG="$HOME/projects/cardano-scripts/src/cscripts/slotLeaderconfig.json"
+LEDGER_FILE="/tmp/ledger.json"
+VRF_KEY_FILE="$HOME/projects/cardano-scripts/src/kaddr_run/vrf.skey"
+CARDANO_LLOG_EXEC="$HOME/projects/cardano-leader-logs/"
+cardano-cli query ledger-state --mainnet --out-file $LEDGER_FILE
```
Comment (dostrelith678) [inline myleaderlog.sh:12]: we don't need to dump ledger state anymore. We can use cardano-cli for querying pool stake and total active stake.

**Review by dostrelith678 — COMMENTED** [review]
I have to test but there are some redundancies

File: cardanoLeaderLogs.js (hunk 1 above)
Comment (nilaysaha) [reply cardanoLeaderLogs.js:28]: I have this after actually installing and running it. So whatever you see in this PR is tested on my server.

Comment (dostrelith678) [reply cardanoLeaderLogs.js:28]: What is `argv[2]` in that case?

File: myleaderlog.sh (hunk 2 above)
Comment (nilaysaha) [reply myleaderlog.sh:12]: ok. Let me change and resubmit PR. Then the argv changes will disappear.

Comment (dostrelith678) [reply myleaderlog.sh:12]: It would be nice if we could test it without the buffer size option, and then add those changes to remove it from the README to this PR as well if it can be removed

**Comment by DamjanOstrelic** [comment]
Closed with #10

**Comment by dostrelith678** [comment]
Closed with #10
